packages/cozy-authentication/MobileRouter.spec.jsx

```diff
@@ -0,0 +1,274 @@
+import { MobileRouter } from './MobileRouter'
+import React from 'react'
+import { shallow } from 'enzyme'
+import { writeState, writeSecret } from './src/utils/onboarding'
+
+describe('MobileRouter', () => {
+  it('should render the appRoutes when all is well', () => {
+    const app = shallow(
+      <MobileRouter
+        appRoutes={<div />}
+        isAuthenticated={true}
+        isRevoked={false}
+        onboarding={{}}
+        onboardingInformations={{}}
+        history={{}}
+        onAuthenticated={jest.fn()}
+        onLogout={jest.fn()}
+        appIcon={''}
+      />
+    )
+    expect(app).toMatchSnapshot()
+  })
+
+  it('should render the appRoutes when no onboarding informations are present', () => {
+    const app = shallow(
+      <MobileRouter
+        appRoutes={<div />}
+        isAuthenticated={true}
+        isRevoked={false}
+        onboarding={{}}
+        onboardingInformations={undefined}
+        history={{}}
+        onAuthenticated={jest.fn()}
+        onLogout={jest.fn()}
+        appIcon={''}
+      />
+    )
+    expect(app).toMatchSnapshot()
+  })
+
+  it('should render the revoked view', () => {
+    const app = shallow(
+      <MobileRouter
+        isAuthenticated={true}
+        isRevoked={true}
+        history={{}}
+        onAuthenticated={jest.fn()}
+        onLogout={jest.fn()}
+        onboarding={{}}
+        onboardingInformations={{}}
+        appIcon={''}
+        appRoutes={<div />}
+      />
+    )
+    expect(app).toMatchSnapshot()
+  })
+
+  it('should render the auth screen when the onboarding has not started', () => {
+    const onboardingInformations = {
+      code: null,
+      state: null,
+      cozy_url: null
```

Comment:
It is unfortunate that the user has to deal with underscores :/ IMHO it would be better if the component did this on its own and what is exposed to the user is cozyURL.